docs/pages/identity-security/session-summaries.mdx

```diff
@@ -0,0 +1,339 @@
+
+Let's start by defining an `inference_model` resource. It tells Teleport how to
+summarize a session by pointing it to a large language model (LLM) API. In
+particular, it contains an API-specific model name. Note that the *Teleport
+inference model name* and *API-specific model name* are two different things.
+This allows, for example, using the same OpenAI model with different
+temperature settings or using different API gateways by creating multiple
+Teleport inference models with the same OpenAI model name.
+
+To communicate with OpenAI, an inference model needs an API key. It's stored
+separately in an `inference_secret` resource and referenced by the
+`inference_model`'s `spec.openai.api_key_secret_ref` field.
+
+<Admonition type="warning">
+To protect your API key, once written, an `inference_secret` value cannot be
+retrieved through `tctl` or Teleport API. It can only be updated or deleted.
+</Admonition>
+
+Finally, we add an `inference_policy` that routes sessions to inference models
+using session kinds and optional custom filters. Only one policy is allowed to
+match a given session to guarantee that its recording will be summarized at
+most once. Policies are checked in alphabetical order of names and the first
+one that matches is used. Here are some example scenarios that can be expressed
+using inference policies:
+
+- Only summarize sessions to your production resources using resource labels.
+- Use different models for summarizing shell and database sessions.
+
+To create the required configuration resources:
+
+1. Create an `inference-config.yaml` file with the following content:
+
+   ```yaml
+   kind: inference_model
+   version: v1
+   metadata:
+     name: shell-summary-model
+   spec:
+     # "openai" is currently the only supported inference provider.
+     openai:
+       # In this scenario, where we connect to OpenAI directly, openai_model_id
+       # needs to be a valid OpenAI model name.
+       openai_model_id: gpt-4o
+       # This value refers to the `inference_secret` resource name.
+       api_key_secret_ref: openai-key
+     # It's a good idea to limit the session length to prevent incurring cost for
+     # sessions that were too long anyway to fit in the model's context window.
+     # If unset, defaults to 200kB. See the Limitations section for more details.
+     max_session_length_bytes: 190000
+
+   ---
+
+   kind: inference_secret
+   version: v1
+   metadata:
+     name: openai-key
+   spec:
+     value: '<paste-your-openai-api-key-here>'
+
+   ---
+
+   kind: inference_policy
+   version: v1
+   metadata:
+     name: shell-sessions
+     description: Summarize all sessions using OpenAI
+   spec:
+     # Session kinds affected by this policy. Allowed values: "ssh", "k8s", "db".
+     kinds:
+     - ssh
+     - k8s
+     - db
+     # Name of the `inference_model` resource that will be used for summarizing
+     # these sessions.
+     model: shell-summary-model
```

Comment:
These resources all seem to reference each other by name.

Just curious if we do any sort of checking to verify that the referenced resource exists and to help catch typos.

Reply:
No, not in this version. :) But I keep that in mind.